### PR TITLE
Add option to opt into full-RBF when sending funds (rebase, original by petertodd)

### PR DESCRIPTION
**Pull request by laanwj**
Rebases #7132
Also adds the comment for https://github.com/bitcoin/bitcoin/pull/7132#discussion_r67606725

Original pull has plenty of ACKs. After all this time we still have no way to opt in to RBF in the wallet, let's not try to bikeshed it to death this time.

### CODE REVIEW

File: src/wallet/wallet.cpp [hunk 1]
```diff
@@ -2355,11 +2356,17 @@ bool CWallet::CreateTransaction(const vector<CRecipient>& vecSend, CWalletTx& wt
 
                 // Fill vin
                 //
-                // Note how the sequence number is set to max()-1 so that the
-                // nLockTime set above actually works.
+                // Note how the sequence number is set to non-maxint so that
+                // the nLockTime set above actually works.
+                /
```
Comment (MarcoFalke) [inline src/wallet/wallet.cpp]: I don't think this is valid cpp :stuck_out_tongue:

**Comment by luke-jr** [comment]
I thought the commandline option was to be renamed to `-walletrbf`?

**Comment by dcousens** [comment]
I also think it might be worth considering the suggestion by @jtimon:

``` c++
fOptIntoFullRbf ? SEQUENCE_OPT_OUT -1 : SEQUENCE_OPT_OUT
```

Instead of 

``` c++
std::numeric_limits<unsigned int>::max() - (fOptIntoFullRbf ? 2 : 1);
```

**Comment by petertodd** [comment]
utACK

**Comment by laanwj** [comment]
> I thought the commandline option was to be renamed to -walletrbf?

Fine with me.

**Comment by paveljanik** [comment]
Why not rename the `DEFAULT_..` as well to match the option name?

**Comment by luke-jr** [comment]
utACK b54c36703c4023114b05305b2edf15db88746fef as-is

http://codepad.org/fhQWSuAk could be included to address @paveljanik's suggestion

**Comment by laanwj** [comment]
Squashed in @luke-jr's patch

File: src/wallet/wallet.h [hunk 2]
```diff
@@ -53,6 +54,8 @@ static const bool DEFAULT_SPEND_ZEROCONF_CHANGE = true;
 static const bool DEFAULT_SEND_FREE_TRANSACTIONS = false;
 //! -txconfirmtarget default
 static const unsigned int DEFAULT_TX_CONFIRM_TARGET = 2;
+//! -walletrbf default
+static const bool DEFAULT_WALLET_RBF = false;
```
Comment (rebroad) [inline src/wallet/wallet.h]: why false by default?

**Comment by MarcoFalke** [comment]
Please see the linked pull in the description

**Pull request merged**